### ISSUE TITLE
Counters probe can be negative (and shouldn't?)

### ISSUE THREAD

**Issue by franz1981** [JCTools/JCTools#372]
@qwwdfsad 

In the current counter code the fall back probe math contains
```java
        int probe = (int) ((Thread.currentThread().getId() * 0x9e3779b9) & Integer.MAX_VALUE);
        // xorshift
        probe ^= probe << 13;
        probe ^= probe >>> 17;
        probe ^= probe << 5;
        return probe;
```
but given that it basically perform a `% (2^31)` and `2^31` is beyond what Java integer signed (SIGH!!!!) types can represent, it can produce negative `probe` values (!!!). Luckly the subsequent mask with the length of the counter array won't make it become a real problem, but still...
Shouldn't be instead?
```java
        int probe = (int) ((Thread.currentThread().getId() * 0x9e3779b9) % 2^30;
```
with `2^30` (or 1 << 30) being the max power of 2 positive integer.

**Comment by franz1981** [comment]
If threadId := 16 I still see probe to become negative due to the xorshift, so maybe we need to turn it into an unsigned int before returning the probe, instead.

**Comment by qwwdfsad** [comment]
Hi!

Sorry, I contributed this 6 years ago and have a very rusty recollection of invariants preserved.

If I recall correctly, fallback is a direct counter-part of `ThreadLocalRandom`. Probe should be an arbitrary random number bound to thread to provide a decent distribution, thus `id * golden_ration_fraction` spread via xorshift. It's definitely expected to have it negative as it's indeed further masked.

Why exactly I was `and`ing it with MAX_VALUE extended on `long` type eludes me.
My guess is that I've messed up the narrowing conversion :) Most likely this `& Integer.MAX_VALUE` is not needed at all
